### ISSUE TITLE
Wendeschaltung Gänge fix / Autohold

### ISSUE THREAD

**Issue by DaMartin121** [modelleicher/FS22_realismAddon_gearbox#10]
Hi Modelleicher, erstmal danke für den tollen Mod!
Ich hätte nur ein kleines Feature Request. Momentan ist es so, wenn man die Wendeschaltung auf Rückwärts stellt, dass das Sequentielle schalten der Gänge genau anders rum funktioniert wie vorwärts. Schaltet man z.B. von -2 hoch, landet man in -1 und nicht in -3 wie man es erwarten würde. Ist bei Lastschaltgetrieben etwas irritierend.
Und wo wir gerade bei Lastschaltern sind, wäre es eventuell möglich eine Autohold Funktion einzubauen? 

LG Martin

**Comment by modelleicher** [comment]
Hi,
Ich glaube nicht dass das durch oder mit dem realismAddon_gearbox kommt dass die Schaltung rückwärts verkehrt herum ist, das ist auch im Standard LS so, oder? An den Standard Inputs habe ich eigentlich nicht vor etwas zu ändern, ist auch nicht ganz so einfach.
Eventuell werde ich aber weitere Custom Inputs einbauen.. Aber in wiefern das in dem Fall hilft ist fraglich.

Was genau ist eine Autohold Funktion? 

LG

**Comment by DaMartin121** [comment]
Ja, das hat nichts mit deinem Mod zu tun. Dachte nur eventuell könntest du dich dem annehmen, da es einfach unlogisches Getriebeverhalten ist. :)

Autohold ist bei Automatik Getrieben weit verbreitet. Bleibt man im Stillstand kurz auf der Bremse, geht das Getriebe automatisch in Neutral und man kann einfach von der Bremse runtergehen und das Fahrzeug hält. Geht man wieder aufs Gas weis das Getriebe man will wieder anfahren.
Das große Problem bei Automatik Getrieben und dem Mod ist momentan, man kann nicht anhalten. Die Automatik legt immer einen Gang ein und durch den Mod ruckelt das Fahrzeug selbst bei getretener Bremse nach vorne. Die Kupplung ist im Automatikmodus auch deaktiviert, also kann man nicht mal auskuppeln.
Das gab es damals im Gearbox Addon und wenn mich nicht alles täuscht auch im 19er VCA.

In den meisten fällen natürlich kein großes Problem das Getriebe einfach im manuellen Modus zu fahren, aber beispielsweise ein 12 Gang Lkw Getriebe ist halt eigentlich nicht dafür gedacht.

**Comment by modelleicher** [comment]
Danke für die Info, ich behalte das mal im Hinterkopf :)

Ja mir waren die manuellen Getriebe und der manuelle Modus in erster Linie am wichtigsten, um den automatischen Modus hab ich mich noch gar nicht gekümmert. Anfangs war ja geplant dass im automatischen Modus das Script komplett deaktiviert ist, allerdings ist irgendwie die Abfrage dafür immer true egal ob manueller oder automatischer Modus.. xD
Dass das Script auch im automatik Modus an ist ist also eher ein Bug eigentlich.. :D

LG

**Comment by jason0611** [comment]
Hallo modelleicher,

müsste die Abfrage zur Erkennung des manuellen Modus in Zeile 134 nicht
`if isManualTransmission and motor.gearShiftMode == VehicleMotor.SHIFT_MODE_MANUAL_CLUTCH then`
statt
`if isManualTransmission and VehicleMotor.SHIFT_MODE_MANUAL_CLUTCH then`
lauten? Nach meinem Verständnis sollte dann eine zuverlässige Erkennung des Getriebemodus möglich sein.

Viele Grüße,
Martin

**Comment by modelleicher** [comment]
> Hallo modelleicher,
> 
> müsste die Abfrage zur Erkennung des manuellen Modus in Zeile 134 nicht `if isManualTransmission and motor.gearShiftMode == VehicleMotor.SHIFT_MODE_MANUAL_CLUTCH then` statt `if isManualTransmission and VehicleMotor.SHIFT_MODE_MANUAL_CLUTCH then` lauten? Nach meinem Verständnis sollte dann eine zuverlässige Erkennung des Getriebemodus möglich sein.
> 
> Viele Grüße, Martin

Hi.. Ja.. das ergibt ne ganze Menge Sinn :D Hab ich total dran vorbei gedacht, Danke!

**Comment by SbSh-DNA-ModDing** [comment]
Hi, mit den SHIFT_MODE_xyz fragt man nur die jeweilige Einstellung ab, also was der Player in seinen eigenen Settings benutzt.
Die Abfrage im Vehicle ist etwas verwirrter.
Ich ich das z.B. so gelöst.

`spec.isVarioTM = self.spec_motorized.motor.lastManualShifterActive == false and self.spec_motorized.motor.groupType == 1 and self.spec_motorized.motor.gearType == 1 and self.spec_motorized.motor.forwardGears == nil`

andersrum für Schalter wäre dann
`spec.isMTM = self.spec_motorized.motor.lastManualShifterActive == true and self.spec_motorized.motor.groupType == 1 and self.spec_motorized.motor.gearType == 1 and self.spec_motorized.motor.forwardGears ~= nil`

groupType und gearType - 1 muss mit Kupplung, 2 geht ohne Kupplung
und dann wäre da noch "gearShiftMode", was das gleiche wie oben SHIFT_MODE_xyz ist 1=Automatik, 2=Manuell, 3 = Manuell mit Kupplung

Das einzige was ich noch nicht habe, z.B. dieser Iseki tvj hat via xml ein Automatik-Schaltgetriebe, der hat grp & gear 1 & 1
das lässt sich so nicht abfrage, da muss man tatsächlich aus der vehicle.xml den transmission.name Wert holen.

Das Iseki Teil kann man nicht anhalten^^
